discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -0,0 +1,305 @@
+                last_tx = transactions_array[-1]
+                last_tx_signature = last_tx["signature"]
+                # Append batch of processed signatures
+                if transaction_signature_batch:
+                    transaction_signatures.append(transaction_signature_batch)
+
+                # Ensure processing does not grow unbounded
+                if len(transaction_signatures) > TX_SIGNATURES_MAX_BATCHES:
+                    logger.info(f"index_solana_plays.py | slicing tx_sigs from {len(transaction_signatures)} entries")
+                    transaction_signatures = transaction_signatures[-TX_SIGNATURES_RESIZE_LENGTH:]
+                    logger.info(f"index_solana_plays.py | sliced tx_sigs to {len(transaction_signatures)} entries")
+
+                # Reset batch state
+                transaction_signature_batch = []
+        logger.info(
+            f"index_solana_plays.py | intersection_found={intersection_found}, last_tx_signature={last_tx_signature}"
+        )
+
+    logger.info(f"index_solana_plays.py | {transaction_signatures}, {len(transaction_signatures)} entries")
+
+    transaction_signatures.reverse()
+
+    logger.info(f"index_solana_plays.py | {transaction_signatures}")
+
+    num_txs_processed = 0
+
+    for tx_sig_batch in transaction_signatures:
+        logger.error(f"index_solana_plays.py | processing {tx_sig_batch}")
+        batch_start_time = time.time()
+        # Process each batch in parallel
+        with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+            with db.scoped_session() as session:
+                parse_sol_tx_futures = {
+                    executor.submit(parse_sol_play_transaction, session,
+                                    solana_client, tx_sig): tx_sig
+                    for tx_sig in tx_sig_batch
+                }
+                for future in concurrent.futures.as_completed(
+                        parse_sol_tx_futures):
+                    try:
+                        # No return value expected here so we just ensure all futures are resolved
+                        future.result()
+                        num_txs_processed += 1
+                    except Exception as exc:
+                        logger.error(f"index_solana_plays.py | {exc}")
+
+        batch_end_time = time.time()
+        batch_duration = batch_end_time - batch_start_time
+        logger.info(
+            f"index_solana_plays.py | processed batch {len(tx_sig_batch)} txs in {batch_duration}s"
+        )
+
+    # Update plays iff some batch is found
+    if num_txs_processed > 0:
+        logger.info(
+            f"index_solana_plays.py | processed {num_txs_processed} txs"
+        )
+        with db.scoped_session() as session:
+            logger.info(
+                f"index_solana_plays.py | Refreshing aggregate_plays materialized view"
+            )
+            session.execute("REFRESH MATERIALIZED VIEW CONCURRENTLY aggregate_plays")
```

Comment:
i dont know if we need to do this every indexing job since the other job will already by doing it. perhaps it makes more sense to either remove there, remove here, or put into a separate job (we have an index_aggregrate_views task)

Reply:
wdyt about putting this into the existing index materialized views task?